File: csharp/resistor-color/ResistorColor.cs
using System;

public static class ResistorColor
{
  private static string[] COLORS = { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };

  public static int ColorCode(string color)
  {
    int index = Array.IndexOf(COLORS, color);
    return index;
  }


  public static string[] Colors()
  {
    return (string[])COLORS.Clone();
  }
}

File: dart/resistor-color/lib/resistor_color.dart
class ResistorColor {
  List<String> colors = [
    'black',
    'brown',
    'red',
    'orange',
    'yellow',
    'green',
    'blue',
    'violet',
    'grey',
    'white'
  ];

  int colorcode(String color) {
    return colors
        .asMap()
        .keys
        .firstWhere((k) => colors[k] == color, orElse: () => null);
  }
}
